cpu/stm32/cpu_common.c

```diff
@@ -51,343 +77,350 @@ static const uint8_t apbmul[] = {
+#endif
+#ifdef AHB22_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB22:
+        return &AHB22_PERIPH_DIS;
+#endif
+#ifdef AHB3_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB3:
+        return &AHB3_PERIPH_DIS;
+#endif
+#ifdef AHB4_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case AHB4:
+        return &AHB4_PERIPH_DIS;
+#endif
+#ifdef IOP_PERIPH_DIS
+    #define RCC_REG_IS_ATOMIC 1
+    case IOP:
+        return &IOP_PERIPH_DIS;
 #endif
-#if defined(CPU_FAM_STM32WL) || defined(CPU_FAM_STM32U5)
-        case APB3:
-            RCC->APB3ENR &= ~(mask);
-            break;
-#endif
-#if defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32WB) || \
-    defined(CPU_FAM_STM32G4) || defined(CPU_FAM_STM32L5) || \
-    defined(CPU_FAM_STM32U5) || defined(CPU_FAM_STM32WL)
-        case APB12:
-            RCC->APB1ENR2 &= ~(mask);
-            break;
-#elif defined(CPU_FAM_STM32G0) || defined(CPU_FAM_STM32C0)
-        case APB12:
-            RCC->APBENR2 &= ~(mask);
-            break;
-#endif
-#if defined(CPU_FAM_STM32L0)
-        case AHB:
-            RCC->AHBENR &= ~(mask);
-            break;
-        case IOP:
-            RCC->IOPENR &= ~(mask);
-            break;
-#elif defined(CPU_FAM_STM32L1) || defined(CPU_FAM_STM32F1) || \
-      defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F3)
-        case AHB:
-            RCC->AHBENR &= ~(mask);
-            break;
-#elif defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || \
-      defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32F7) || \
-      defined(CPU_FAM_STM32WB) || defined(CPU_FAM_STM32G4) || \
-      defined(CPU_FAM_STM32L5) || defined(CPU_FAM_STM32U5) || \
-      defined(CPU_FAM_STM32WL)
-        case AHB1:
-            RCC->AHB1ENR &= ~(mask);
-            break;
-/* STM32F410 RCC doesn't provide AHB2 and AHB3 */
-#if !defined(CPU_LINE_STM32F410Rx)
-        case AHB2:
-#if defined(CPU_FAM_STM32U5)
-            RCC->AHB2ENR1 &= ~(mask);
+
+    /* If MCU does not have separate set/clear bits. */
+#if RCC_REG_IS_ATOMIC == 0
+    default:
+        return _rcc_en_reg(bus);
 #else
-            RCC->AHB2ENR &= ~(mask);
-#endif
-            break;
-#if defined(CPU_FAM_STM32U5)
-        case AHB22:
-            RCC->AHB2ENR2 &= ~(mask);
-            break;
+    case BUS_NUMOF:
+        assert(false);
+        return NULL;
 #endif
-        case AHB3:
-            RCC->AHB3ENR &= ~(mask);
-            break;
+    }
+
+    assert(false);
+    return NULL;
+}
+
+static volatile uint32_t* _rcc_lp_en_reg(bus_t bus)
+{
+    /* Note this switch case is designed in such a way that a default is only
+       provided when all other cases are *not* provided. This is to ensure that
+       either all cases or no cases are provided. Anything else will cause the
+       compiler to emit a warning. */
+    switch (bus) {
+#ifdef APB1_PERIPH_LP_EN
+    #define HAS_LP_MODE 1
+    case APB1:
+        return &APB1_PERIPH_LP_EN;
+#endif
+#ifdef APB12_PERIPH_LP_EN
+    #define HAS_LP_MODE 1
```

Comment:
seems like this can be defined in line 37 ?

Reply:
`#define HAS_LP_MODE 1`

Reply:
I consider doing it that way during development, but ultimately decided that it must be done as I did it to avoid potential for human error when porting to new processors. The goal was to minimize the amount of code one must touch or understand to extend support to a new MCU.